ambry-utils/src/test/java/com.github.ambry.utils/SimpleByteBufferPoolTest.java

```diff
@@ -130,13 +148,18 @@ public void testFirstBlockedThenSucceed()
       used[i] = new CountDownLatch(1);
       consumers[i] = new BufferConsumer(size, maxBlockTimeInMs, pool, allocated[i], used[i]);
       consumerThreads[i] = new Thread(consumers[i]);
-      consumerThreads[i].start();
     }
-    allocated[0].await();
+    consumerThreads[0].start();
+    if (!allocated[0].await(100, TimeUnit.MILLISECONDS)) {
+      fail("SimpleByteBufferPool takes too long time to allocate a buffer to BufferConsumer[0].");
+    }
     assertEquals(0, pool.availableMemory());
+    consumerThreads[1].start();
```

Comment:
what is being tested here? how is it being verified that the second thread actually blocks?